### PR TITLE
Backport fix for containerd

### PR DESCRIPTION
**Pull request by ncopa**
https://github.com/containerd/containerd/pull/7845

Signed-off-by: Natanael Copa <ncopa@mirantis.com>

## Description

Fixes https://github.com/containerd/containerd/issues/7843

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [ ] Auto test added

## Checklist:

- [ ] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [ ] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: embedded-bins/containerd/Dockerfile [hunk 1]
```diff
@@ -6,13 +6,15 @@ ENV GOPATH=/go
...
 	btrfs-progs-dev btrfs-progs-static \
-	protoc
+	protoc curl
 
 ARG VERSION
 RUN mkdir -p $GOPATH/src/github.com/containerd/containerd
 RUN git -c advice.detachedHead=false clone -b v$VERSION --depth=1 https://github.com/containerd/containerd.git $GOPATH/src/github.com/containerd/containerd
 WORKDIR /go/src/github.com/containerd/containerd
 
+RUN curl -L https://github.com/containerd/containerd/pull/7845.patch | git apply
```
Comment (juanluisvaladas) [inline embedded-bins/containerd/Dockerfile:16]: I would use the commit instead as the pull request may change and make things unpredictable:
https://github.com/containerd/containerd/commit/cf6bb0841b6ae2cd91255a4df89aee6ea5192dc6.patch

Also I would wait for it to be merged, becuse some of the failures (including a race condition) seem legit to me, but overall I like the idea.

Comment (ncopa) [reply embedded-bins/containerd/Dockerfile:16]: Yes. We shouldn't merge this until its merged upstream. That's why I created it as a draft. But it gives us a possibility to test it early.

**Comment by juanluisvaladas** [comment]
@ncopa I don't know if you saw it but the PR is merged in containerd

**Comment by ncopa** [comment]
I have verified that the is in upstream main branch. https://github.com/containerd/containerd/commit/3ee6dd5c1bca441d1ec4988cbaebadbfbcfde525

**Comment by ncopa** [comment]
upstream did new release: https://github.com/containerd/containerd/releases/tag/v1.6.15

**Comment by ncopa** [comment]
commit in upstream 1.6 branch: https://github.com/containerd/containerd/commit/f16447e2d495d77c03c6644e78877cd3596ef523

**Pull request closed without merging**